<launch>
    <!-- 启动 livox_ros_driver2 的 msg_HAP.launch -->
    <include file="$(find livox_ros_driver2)/launch_ROS1/msg_HAP.launch" />
   
    <!-- 是否启动 RViz 显示 -->
    <arg name="rviz" default="true" />
 
    <!-- 加载 HAP 的配置文件 -->
    <rosparam command="load" file="$(find fast_lio_localization)/config/hap_tx.yaml" />
 
    <!-- 设置参数值 -->
    <param name="feature_extract_enable" type="bool" value="0"/>
    <param name="point_filter_num" type="int" value="3"/>
    <param name="max_iteration" type="int" value="3" />
    <param name="filter_size_surf" type="double" value="0.5" />
    <param name="filter_size_map" type="double" value="0.5" />
    <param name="cube_side_length" type="double" value="1000" />
    <param name="runtime_pos_log_enable" type="bool" value="0" />
    <param name="pcd_save_enable" type="bool" value="0" />
    
    <!-- 启动激光雷达映射节点 -->
    <node pkg="fast_lio_localization" type="fastlio_mapping" name="laserMapping" output="screen" />

    <!-- 指定地图文件路径 -->
	<arg name="map" default="$(find fast_lio_localization)/PCD/map.pcd" />
	
    <!-- 全局定位节点 -->
    <node pkg="fast_lio_localization" type="global_localization.py" name="global_localization" output="screen" />

	<!-- 变换融合节点 -->
    <node pkg="fast_lio_localization" type="transform_fusion.py" name="transform_fusion" output="screen" />

	<!-- 地图发布节点 -->
    <node pkg="pcl_ros" type="pcd_to_pointcloud" name="map_publishe" output="screen"
		  args="$(arg map) 5 _frame_id:=map cloud_pcd:=/map" />
 
    <!-- 可选：RViz可视化显示 -->
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio_localization)/rviz_cfg/localization.rviz" />
    </group>
 
</launch> 